python/python/lance/dataset.py

```diff
@@ -1486,6 +1494,10 @@ class DatasetOptimizer:
     def __init__(self, dataset: LanceDataset):
         self._dataset = dataset
 
+    def __call__(self, *args, **kwargs):
+        self.compact_files()
+        self.optimize_indices()
```

Comment:
Should we do `dataset.optimize.all()`, make the call more fluent in english.?

which means we have a `DatasetOptimizer.all()` method.

Reply:
Well I plan to add more methods, some of which will require parameters from the user and are mutually exclusive with other optimizations. So I'm not sure we should have an "all" method.

For example, after stats we'll likely add a `dataset.optimize.zorder(cluster_by=["col_a", "col_b"])` (so that stats are more useful). But this command makes `compact_files()` unnecessary, since it will consolidate files as it runs.

Reply:
Ok, lemme remove `all()` from this PR

Reply:
My worry is that we expose too much details, without a reasonable default to do something very simple (from user's perception). Means that users need to understand the implication of why / when to call each call.

Reply:
> My worry is that we expose too much details, without a reasonable default to do something very simple (from user's perception). Means that users need to understand the implication of why / when to call each call.

I agree we should create good defaults and make them easy to call. I've created an issue with some ideas of what we can do there: https://github.com/lancedb/lance/issues/1414